File: src/app/models/Room.js


// Define collection and schema for Room
export interface Room {
    name:
	type : string
    Floor:
	type : Schema.Types.ObjectId
    Devices:
 	type : [{ type: Schema.Types.ObjectId, ref: 'IoTDevice' }]
    Gateways:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Gateway' }]
#
    collection: 'rooms'
}

File: iotonaspdotnet/Domain/Room.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class Room
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long RoomId { get; set; }
 public virtual string Name { get; set; }
public virtual Floor Floor { get; set; }
public virtual IoTDevice Devices { get; set; }
public virtual Gateway Gateways { get; set; }

    public static Room FromRequest(RoomRequest request) {
        return new Room {
            Id = request.Id,
            Name = request.Name,
        };
    }
}
